java-squid/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -217,11 +220,15 @@ public void visitMethodInvocation(MethodInvocationTree syntaxNode) {
           }
         }
       } else {
-        final List<SymbolicValue> values = programState.peekValues(syntaxNode.arguments().size() + 1);
-        final List<SymbolicValue> argumentValues = values.subList(1, values.size());
-        for (SymbolicValue target : argumentValues) {
-          programState = closeResource(programState, target);
-        }
+        attemptToCloseArguments(syntaxNode.arguments(), 1);
+      }
+    }
+
+    private void attemptToCloseArguments(final Arguments arguments, int stackOffset) {
```

Comment:
Why is that an attempt ?  I am not sure of the stackOffset argument, it appears like a magic number to me.